### PR TITLE
VZ-4626 More VMO logging cleanup

### PR DESCRIPTION
**Pull request by pfmackin**
Do more VMO logging cleanup.  Focus mainly on error messages.

### CODE REVIEW

**Review by michael-cico — DISMISSED** [review]

File: pkg/vmo/configmap.go [hunk 1]
```diff
@@ -97,35 +96,32 @@ func CreateConfigmaps(controller *Controller, vmo *vmcontrollerv1.VerrazzanoMoni
...
 	if err != nil {
-		zap.S().Errorf("Failed to create configmap %s for reason %v", vmo.Spec.Prometheus.VersionsConfigMap, err)
 		return err
 
 	}
 	configMaps = append(configMaps, vmo.Spec.Prometheus.VersionsConfigMap)
 
 	// Delete configmaps that shouldn't exist
-	zap.S().Debug("Deleting unwanted ConfigMaps for vmo '%s' in namespace '%s'", vmo.Name, vmo.Namespace)
+	controller.log.Debug("Deleting unwanted ConfigMaps for VMI %s/%s", vmo.Namespace, vmo.Name)
```
Comment (mgianatagh) [inline pkg/vmo/configmap.go:113]: Should this be .Debugf?

File: pkg/vmo/deployment.go [hunk 2]
```diff
@@ -47,7 +47,7 @@ func CreateDeployments(controller *Controller, vmo *vmcontrollerv1.VerrazzanoMon
 			runtime.HandleError(errors.New("deployment name must be specified"))
 			return true, nil
 		}
-		zap.S().Debugf("Applying Deployment '%s' in namespace '%s' for vmo '%s'\n", deploymentName, vmo.Namespace, vmo.Name)
+		zap.S().Debugf("Applying Deployment '%s' in namespace '%s' for VMI '%s'\n", deploymentName, vmo.Namespace, vmo.Name)
```
Comment (stevshap) [inline pkg/vmo/deployment.go:50]: should this be controller.log?

File: pkg/vmo/rolebindings.go [hunk 3]
```diff
@@ -52,7 +52,7 @@ func CreateRoleBindings(controller *Controller, vmo *vmcontrollerv1.VerrazzanoMo
 	}
 
 	// Delete RoleBindings that shouldn't exist
-	zap.S().Debugf("Deleting unwanted RoleBindings for vmo '%s' in namespace '%s'", vmo.Name, vmo.Namespace)
+	zap.S().Debugf("Deleting unwanted RoleBindings for VMI '%s' in namespace '%s'", vmo.Name, vmo.Namespace)
```
Comment (stevshap) [inline pkg/vmo/rolebindings.go:55]: same q, controller.log?

**Review by stevshap — APPROVED** [review]

File: pkg/vmo/service.go [hunk 4]
```diff
@@ -54,10 +53,10 @@ func CreateServices(controller *Controller, vmo *vmcontrollerv1.VerrazzanoMonito
 		}
 
 		if err != nil {
-			zap.S().Errorf("Failed to apply Service for vmo: %s", err)
+			controller.log.Errorf("Failed to apply Service for VMI %s: %v", vmo.Name, err)
```
Comment (mgianatagh) [inline pkg/vmo/service.go:56]: There appear to be some inconsistencies of using "%v" versus "%+v".  For example, lines 47 and 56.

File: pkg/vmo/configmap.go (hunk 1 above)
Comment (pfmackin) [reply pkg/vmo/configmap.go:113]: i can make it once

File: pkg/vmo/pvc.go [hunk 5]
```diff
@@ -59,7 +59,7 @@ func createPersistentVolumeClaims(controller *Controller, vmo *vmcontrollerv1.Ve
 			// We choose to absorb the error here as the worker would requeue the
 			// resource otherwise. Instead, the next time the resource is updated
 			// the resource will be queued again.
-			runtime.HandleError(fmt.Errorf("PVC name must be specified"))
+			runtime.HandleError(fmt.Errorf("Failed, PVC name must be specified"))
```
[the remark this answers is not in the document]
Comment (pfmackin) [reply pkg/vmo/pvc.go:62]: there is no fmt.Error, i will use errors.New

File: pkg/vmo/service.go (hunk 4 above)
Comment (pfmackin) [reply pkg/vmo/service.go:56]: %v should be used i will fix

File: pkg/vmo/statefulset.go [hunk 6]
```diff
@@ -121,14 +119,14 @@ func updateOwnerForPVCs(controller *Controller, statefulSet *appsv1.StatefulSet,
...
-			zap.S().Errorf("Failed to update the owner reference in PVC %s, for the reason (%v)", pvc.Name, err)
+			controller.log.Errorf("Failed to update the owner reference in PVC %s: %v", pvc.Name, err)
 			return err
 		}
 	}
 	expectedNumPVCs := int(*statefulSet.Spec.Replicas) * len(statefulSet.Spec.VolumeClaimTemplates)
 	actualNumPVCs := len(existingPvcList)
 	if actualNumPVCs != expectedNumPVCs {
-		return fmt.Errorf("PVC owner reference set in %v of %v PVCs for VMO %s", actualNumPVCs, expectedNumPVCs, vmoName)
+		return fmt.Errorf("Failed, PVC owner reference set in %v of %v PVCs for VMI %s", actualNumPVCs, expectedNumPVCs, vmoName)
```
[the remark this answers is not in the document]
Comment (pfmackin) [reply pkg/vmo/statefulset.go:129]: %v is ok

**Review by mgianatagh — APPROVED** [review]

**Review by stevshap — APPROVED** [review]

**Pull request merged**